src/Doctrine/Common/State/PersistProcessor.php

```diff
@@ -52,10 +52,34 @@ public function process(mixed $data, Operation $operation, array $uriVariables =
         // https://github.com/doctrine/orm/issues/8461#issuecomment-1250233555
         if ($operation instanceof HttpOperation && HttpOperation::METHOD_PUT === $operation->getMethod() && ($operation->getExtraProperties()['standard_put'] ?? false)) {
             \assert(method_exists($manager, 'getReference'));
-            // TODO: the call to getReference is most likely to fail with complex identifiers
```

Comment:
this is a special case, not yet working even in your patch and is about composite identifiers (multiple identifiers)

Reply:
I haven't tested composite identifiers, but i thought calling getReference with a correct $ormIds array would do the trick, sorry about that. I'll look into this too.